tests/MongoDB.Driver.Tests/CustomServerSelectorTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.Clusters;
+using MongoDB.Driver.Core.Clusters.ServerSelectors;
+using MongoDB.Driver.Core.Configuration;
+using MongoDB.Driver.Core.Events;
+using MongoDB.Driver.Core.Servers;
+using Xunit;
+
+namespace MongoDB.Driver.Tests
+{
+    public class CustomServerSelectorTests
+    {
+        [Fact]
+        public void Should_call_custom_server_selector()
+        {
+            var eventCapturer = new EventCapturer()
+                .Capture<ClusterSelectingServerEvent>()
+                .Capture<ClusterSelectedServerEvent>();
+            var customServerSelector = new CustomServerSelector();
+            var client = DriverTestConfiguration.CreateDisposableClient(
+                clientSettings =>
+                    clientSettings.ClusterConfigurator =
+                        c =>
+                        {
+                            c.ConfigureCluster(
+                                s =>
+                                    new ClusterSettings(
+                                        serverSelectionTimeout: TimeSpan.FromSeconds(2),
+                                        postServerSelector: customServerSelector));
+                            c.Subscribe(eventCapturer);
+                        });
+
+            var collection = client
+                .GetDatabase(DriverTestConfiguration.DatabaseNamespace.DatabaseName)
+                .GetCollection<BsonDocument>(DriverTestConfiguration.CollectionNamespace.CollectionName)
+                .WithReadPreference(ReadPreference.Nearest);
```

Comment:
I would say that this option doesn't make sense here, I would suggest to remove it and focus only on the test target.

Reply:
I implemented this test according to specification: https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection-tests.rst#application-provided-server-selector
So unless there's an issue with this option, I would leave it as is.